File: python/problem6.py
def findSumSquareDifference2(limit):
    sumSquare = ((limit*(limit+1))//2)**2
    squareSum = sum(list([x *x for x in range(1, limit+1)]))
    
    #sq_list = [x * x for x in range(1, limit + 1)]
    #squareSum = sum(item for item in sq_list)
    
    return sumSquare-squareSum

def findSumSquareDifference(limit):
    sumSquare = ((limit*(limit+1))//2)**2
    squareSum = 0
    for x in range(1,limit+1):
        squareSum += x*x
    return sumSquare-squareSum

if __name__ == '__main__':
    print(findSumSquareDifference(10))
    print(findSumSquareDifference2(10))

File: go/problem6.go
package main

import (
	"fmt"
)

func main() {
	fmt.Println(FindSumSquareDifference(10))
}

func FindSumSquareDifference(limit int) int {

	sum := (limit * (limit + 1)) / 2
	sumSquare := sum * sum
	squareSum := 0
	for i := 1; i <= limit; i++ {
		squareSum += i * i
	}
	return sumSquare - squareSum

}
